src/perform_step/predcorr.jl

```diff
@@ -0,0 +1,70 @@
+@muladd function perform_step!(integrator,cache::PCEulerConstantCache,f=integrator.f)
+  @unpack t,dt,uprev,u,W,p,f,g = integrator
+  @unpack theta,eta,bbprime = integrator.alg
+  dW = W.dW
+  if !is_diagonal_noise(integrator.sol.prob) || typeof(W.dW) <: Number
+    gdW_n = g(uprev,p,t)*dW
+  else
+    gdW_n = g(uprev,p,t).*dW
+  end
+
+  f_n = f(uprev,p,t)
+  ubar = @muladd uprev + dt*f_n + gdW_n
+
+  fbar_n = @muladd f_n - eta*bbprime(uprev, p, t)
+
+  tnp1 = t + dt
+  fbar_np1 = @muladd f(ubar, p, tnp1) - eta*bbprime(ubar, p, tnp1)
+
+  if !is_diagonal_noise(integrator.sol.prob) || typeof(dW) <: Number
+    gdW_np1 = g(ubar,p,tnp1)*dW
+  else
+    gdW_np1 = g(ubar,p,tnp1).*dW
+  end
+
+  u .= @muladd uprev + (theta*dt)*fbar_np1 + ((1-theta)*dt)*fbar_n + eta*gdW_np1 + (1-eta)*gdW_n
```

Comment:
You don't want to `.=` for numbers and static arrays. It'll work but be slower.

Reply:
Sure, I will copy what you did for EM then.